crates/volta-core/src/project/mod.rs

```diff
@@ -147,10 +147,17 @@ impl Project {
     pub fn has_direct_bin(&self, bin_name: &OsStr) -> Fallible<bool> {
         if let Some(name) = bin_name.to_str() {
             let config_path = volta_home()?.default_tool_bin_config(name);
-            if config_path.exists() {
-                let config = BinConfig::from_file(config_path)?;
-                return Ok(self.has_direct_dependency(&config.package));
-            }
+
+            return match BinConfig::from_file(config_path) {
+                Err(error) => {
+                    if error.is_not_found_error_kind() {
+                        Ok(false)
+                    } else {
+                        return Err(error);
+                    }
+                }
```

Comment:
This is a fair amount of extra boilerplate that is duplicated in a lot of spots. Perhaps we could create an extra method on `BinConfig` and `PackageConfig` that handles the not found error and returns a `Fallible<Option<ConfigType>>`? Then this section could be condensed to:

```rust
match BinConfig::from_file_if_exists(config_path)? {
    Some(config) => Ok(self.has_direct_dependency(&config.package)),
    None => Ok(false),
}
```

Reply:
Done